### ISSUE TITLE
做内容替换

### ISSUE THREAD

**Issue by lufb** [intel/hyperscan#190]
我想用这个库做内容替换,如果我在hs_scan的回调中,直接将内容替换掉,会不会导致正则匹配有问题

**Comment by xiangwang1** [comment]
Yes, the results are not guaranteed if you modified your input string at the middle of matching.

**Comment by ghcr** [comment]
Btw: Its always a good idea to write in English rather than in Chinese.
